### DISCUSSION TITLE
[@types/eslint] `Linter.configType` should also accept `eslintrc` as possible value.

### DISCUSSION THREAD

**Discussion by virtuallyunknown** [DefinitelyTyped/DefinitelyTyped#69349] [Issues with a @types package] [answered]
Greetings!

ESLint recently shipped version 9.0.0, and [as described in the docs](https://github.com/eslint/eslint/blob/b7cf3bd29f25a0bab4102a51029bf47c50f406b5/docs/src/use/migrate-to-9.0.0.md?plain=1#L697) and [the source code](https://github.com/eslint/eslint/blob/main/lib/linter/linter.js#L1304), the `Linter` class now also accepts `eslintrc` as a possible value for`configType` in it's constructor parameters.

So with that said, I think the updated implementation should be a literal union:

```ts
export class Linter {
    // ...
    constructor(options?: { cwd?: string | undefined; configType?: "flat" | "eslintrc" });
    // ...                                                                ^^^^^^^^^^^^
}
```

Relevant code here:

- https://github.com/DefinitelyTyped/DefinitelyTyped/blob/0cebe5d5bcaad4a8f18e525b1cd098583b252585/types/eslint/index.d.ts#L844C55-L844C74

Relevant issue:

- https://github.com/eslint/eslint/issues/18322#issuecomment-2053713718

Also, please note that support for `eslintrc` configs [will be removed in version 10 of ESLint.](https://github.com/eslint/eslint/issues/18322#issuecomment-2053746038)

Cheers!

**Comment by ghost** [comment, marked as answer]
Fixed in #69368

**Reply by virtuallyunknown** [reply]
Cheers!
